native/cocos/platform/openharmony/modules/CanvasRenderingContext2DDelegate.cpp

```diff
@@ -240,31 +267,39 @@ void CanvasRenderingContext2DDelegate::fillTextureData() {
 }
 
 std::array<float, 2> CanvasRenderingContext2DDelegate::convertDrawPoint(Point point, const std::string &text) {
-    Size textSize = measureText(text);
+    auto typography = createTypography(text);
+    Size textSize {static_cast<float>(OH_Drawing_TypographyGetMaxIntrinsicWidth(typography->get())),
+                   static_cast<float>(OH_Drawing_TypographyGetHeight(typography->get()))};
     if (_textAlign == CanvasTextAlign::CENTER) {
         point[0] -= textSize[0] / 2.0f;
     } else if (_textAlign == CanvasTextAlign::RIGHT) {
         point[0] -= textSize[0];
     }
-
+    double alphabeticBaseLine = OH_Drawing_TypographyGetAlphabeticBaseline(typography->get());
     if (_textBaseLine == CanvasTextBaseline::TOP) {
-#if defined(OPENHARMONY_DRAW)
-        // DrawText default
-        GetTextMetrics(_DC, &_tm);
-        point[1] += -_tm.tmInternalLeading;
-#endif
+        //point[1] += -alphabeticBaseLine;
     } else if (_textBaseLine == CanvasTextBaseline::MIDDLE) {
         point[1] += -textSize[1] / 2.0f;
     } else if (_textBaseLine == CanvasTextBaseline::BOTTOM) {
         point[1] += -textSize[1];
     } else if (_textBaseLine == CanvasTextBaseline::ALPHABETIC) {
         //GetTextMetrics(_DC, &_tm);
         //point[1] -= _tm.tmAscent;
+        point[1] -= alphabeticBaseLine;
     }
-
     return point;
 }
 
+std::unique_ptr<CanvasRenderingContext2DDelegate::ScopedTypography> CanvasRenderingContext2DDelegate::createTypography(const std::string &text) {
+    OH_Drawing_TypographyHandlerPushTextStyle(_typographyCreate, _textStyle);
```

Comment:
When measuring text, it needs to be called, and when drawing text, it needs to be created.
I tested it, if I don't do that, there will be problems with the text display. 
I think the typography needs to be created when the text changes.